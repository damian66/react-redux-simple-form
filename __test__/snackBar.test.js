import React from 'react'
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import ConnectedSnackBar, { SnackBar } from './../client/src/components/SnackBar'
import configureStore from 'redux-mock-store'
import {Provider} from 'react-redux'

import {createStore} from 'redux'

// Snapshot
describe('>>>SnackBar --- Snapshot',()=>{    
    it('+++capturing Snapshot of SnackBar', () => {
        const renderedValue =  renderer.create(<SnackBar />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});

/* ***** ***** ***** */

describe('>>>SnackBar --- Shallow Render REACT COMPONENTS',()=>{
    const initialState = {
        msg: "Test",
        state: 1
    }
    const mockStore = configureStore()
    let store, wrapper

    beforeEach(()=>{
        store = mockStore(initialState)
        wrapper = mount( <Provider store={store}><ConnectedSnackBar /></Provider> )        
    })

    it('+++ render the component', () => {
       expect(wrapper.length).toEqual(1);
    })
});
