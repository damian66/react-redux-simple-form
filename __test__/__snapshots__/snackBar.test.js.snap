// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>SnackBar --- Snapshot +++capturing Snapshot of SnackBar 1`] = `
<div
  className="snackBar "
/>
`;
